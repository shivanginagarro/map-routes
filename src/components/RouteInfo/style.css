.route-info-wrapper{
    width: 30 + "%";
    display: flex;
    float: left;
}