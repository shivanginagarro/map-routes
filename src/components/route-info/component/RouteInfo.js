import React from "react";
import PropTypes from 'prop-types';

//its is more like a  presentational component for displaying the route related information from backend.  

const RouteInfo = (props) => {
    const { informationDetail } = props;
    return (
        <>
        {informationDetail.total_distance ? <div><label className={"label-header"}>Total Distance</label>:{informationDetail.total_distance}</div> : null}
        {informationDetail.total_time ? <div><label className={"label-header"}>Total Time</label>:{informationDetail.total_time}</div> : null}
        {informationDetail.error ? <div className={"inline-error"}><label className={"label-header"}>Error</label>:{informationDetail.error}</div> : null}
        </>
    )
}


RouteInfo.protoType = {
    informationDetail : PropTypes.object.isRequired,
}
export default RouteInfo;