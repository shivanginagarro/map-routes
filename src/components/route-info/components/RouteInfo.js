import React from "react";

const RouteInfo = (props) => {
    const { informationDetail } = props;
    return (
        <>
        {informationDetail.total_distance ? <div><label>Total Distance</label>:{informationDetail.total_distance}</div> : null}
        {informationDetail.total_time ? <div><label>Total Time</label>:{informationDetail.total_time}</div> : null}
        {informationDetail.error ? <div><label>Error</label>:{informationDetail.error}</div> : null}
        </>
    )
}

export default RouteInfo;