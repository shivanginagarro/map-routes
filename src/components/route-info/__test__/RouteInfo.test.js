import React from "react";
import RouteInfo from "../component/RouteInfo";
import renderer from "react-test-renderer";


describe("Component:RouteInfo",() => {
   it("Testing the Route Info Component",() => {
    let informationDetail = {
        error: '',
        total_distance: '',
        total_time: ''
    }
    let tree = renderer.create(<RouteInfo informationDetail={informationDetail} />).toJSON();
    expect(tree).toMatchSnapshot();
   });
})