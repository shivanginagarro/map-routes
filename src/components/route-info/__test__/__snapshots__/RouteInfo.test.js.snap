// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:RouteInfo Testing the Route Info Component 1`] = `null`;
